From 8c5ac056736db88aef94f3002eb727a9dc22f7a4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fr=C3=A9d=C3=A9ric=20L=C3=A9caille?= <flecaille@haproxy.com>
Date: Wed, 12 Apr 2023 13:41:54 +0200
Subject: [PATCH] MINOR: quic: Add connection flags to traces

This should help in diagnosing issues.

Must be backported to 2.7 and 2.6.
---
 src/quic_conn.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/quic_conn.c b/src/quic_conn.c
index 9412b8900..1670da40f 100644
--- a/src/quic_conn.c
+++ b/src/quic_conn.c
@@ -280,7 +280,7 @@ static void quic_trace(enum trace_level level, uint64_t mask, const struct trace
 	if (qc) {
 		const struct quic_tls_ctx *tls_ctx;
 
-		chunk_appendf(&trace_buf, " : qc@%p", qc);
+		chunk_appendf(&trace_buf, " : qc@%p flags=0x%x", qc, qc->flags);
 		if (mask & QUIC_EV_CONN_INIT) {
 			chunk_appendf(&trace_buf, "\n  odcid");
 			quic_cid_dump(&trace_buf, &qc->odcid);
@@ -5673,6 +5673,7 @@ void quic_conn_release(struct quic_conn *qc)
 
 	pool_free(pool_head_quic_conn_rxbuf, qc->rx.buf.area);
 	pool_free(pool_head_quic_conn, qc);
+	qc = NULL;
 
 	TRACE_PROTO("QUIC conn. freed", QUIC_EV_CONN_FREED, qc);
 
